syz-manager/snapshot.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+	"unsafe"
+
+	"github.com/google/flatbuffers/go"
+	"github.com/google/syzkaller/pkg/flatrpc"
+	"github.com/google/syzkaller/pkg/fuzzer/queue"
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/vm"
+)
+
+func (mgr *Manager) snapshotLoop() {
+	queue.StatNumFuzzing.Add(mgr.vmPool.Count())
+
+	reqc := make(chan *queue.Request, 2*mgr.vmPool.Count())
```

Comment:
This might not work well with the VM distribution logic from your other PR. Yes, we don't need it in the case of snapshotting that much (though differently booted VMs may still give some more variety?).

I'd just go with calling `mgr.source.Next()` directly from `snapshotVM()`. The cost of generating an input is tiny compared to the snapshot rollbacks and, except for `exec gen/fuzz`, we already create execution requests concurrently.

Reply:
Removed the channel.